modules/oe_theme_helper/src/Plugin/Field/FieldFormatter/FeaturedMediaRenderedEntityFormatter.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_theme_helper\Plugin\Field\FieldFormatter;
+
+use Drupal\Core\Field\FieldItemInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\Plugin\Field\FieldFormatter\EntityReferenceFormatterBase;
+use Drupal\file\Entity\File;
+use Drupal\media\MediaInterface;
+
+/**
+ * Plugin implementation of the 'Featured media as entity' formatter.
+ *
+ * @FieldFormatter(
+ *   id = "oe_featured_media_entity_view",
+ *   label = @Translation("Rendered entity"),
+ *   description = @Translation("Display the entity of the referenced media and the caption."),
+ *   field_types = {
+ *     "oe_featured_media"
+ *   }
+ * )
+ */
+class FeaturedMediaRenderedEntityFormatter extends EntityReferenceFormatterBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $elements = [];
+
+    foreach ($items as $delta => $item) {
+      $elements[$delta] = $this->viewElement($item);
+    }
+
+    return $elements;
+  }
+
+  /**
+   * Renders a single field item.
+   *
+   * @param \Drupal\Core\Field\FieldItemInterface $item
+   *   The individual field item.
+   *
+   * @return array
+   *   The ECL media-container parameters.
+   */
+  protected function viewElement(FieldItemInterface $item): array {
+    $params = [];
+    $params['description'] = $item->caption;
+
+    if (!$item->entity instanceof MediaInterface) {
```

Comment:
Since this formatter only applies to a specific field type we can assume that the entity is a media.

Reply:
We need to ensure that media entity exists as it can be removed outside. The same approach as https://github.com/openeuropa/oe_theme/pull/513/files#diff-175fc9d25ec415590a71139896c6b64dR31 is used here.

Reply:
I tested attaching and then deleting a media, after deletion $item->entity will not be MediaInterface and we get errors, so this has to stay